### ISSUE TITLE
Use ctrl+click to extend selection

### ISSUE THREAD

**Issue by sylvain-hamel** [cytoscape/cytoscape.js#973]
Cytoscape uses `shift-click` to extend selection. This is not a problem per se but it should also support `ctrl+click` to do so.

In general, `ctrl+click` is the common way to add individual elements to the current selection. `Shift-click` is normally used to add "all items between two elements" to a selection, which is not as common of a use case.

I've had several users test my application and when told to *"Select two elements and drag them"*, they all tried to use `ctrl+click` to do it. None of them figured out that they had to use `shift-click`. I had to tell them.

Can you please bind `ctrl+click` to extend selection.

**Comment by maxkfranz** [comment]
Cross platform apps use `Shift` for multiple selection, like Photoshop.  `Ctrl` doesn't work well across platforms, especially on web.  It works fine on Windows, but on Mac the analogue of `Ctrl` is `Command`.  Supporting both `Ctrl` and `Command` doesn't work:

- `Command` is `Meta` on Windows and opens the start menu.
- `Ctrl` on Mac with click opens the context menu.

**Comment by sylvain-hamel** [comment]
If you don't want to have this "on" by default, can you add an option for that (`useCtrlTapToExtendSelection : true` or something like this)? Because this is a must-have for me.

**Comment by sylvain-hamel** [comment]
Or you could detect the platform using the User Agent to enable it.

**Comment by sylvain-hamel** [comment]
By the way, in my case the web browser is embedded in a Windows native application using https://github.com/cefsharp/CefSharp so I'm sure `ctrl` will do what I need.

**Comment by sylvain-hamel** [comment]
Hi @maxkfranz, will you add the option to support `ctrl+click`? I'm still getting bug reports from users who claim they can't multi-select elements (because they intuitively do `ctrl+click` instead of `shift+click`).

**Comment by maxkfranz** [comment]
You can try this, but I can't guarantee it will make it into 2.5.  If it's buggy, it will have to be removed.

Worst case, you'll have to fork if this is reverted.

**Comment by sylvain-hamel** [comment]
Thanks a lot. I'm going to try it right away.

**Comment by sylvain-hamel** [comment]
Hi, I tried it and it works as I expected. Selecting multiple items using `ctrl+drag` still works fine too. 

I read your comment in the code regarding the alt Key. I don't think you should map `alt+click` because this is not normally used for selection (at least not on Windows).

Let me know if I can help in any way.

**Comment by megawebmaster** [comment]
Anything new on this matter? Is it going to be merged into master?

**Comment by maxkfranz** [comment]
Once 2.5 is complete

**Comment by maxkfranz** [comment]
I'm going to close this for now unless a bug is found with this feature

**Comment by megawebmaster** [comment]
Thank you.
